server/filestore.go

```diff
@@ -8518,6 +8523,7 @@ func (mb *msgBlock) recalculateForSubj(subj string, ss *SimpleState) {
 			li := int(bi) - mb.cache.off
 			if li >= len(mb.cache.buf) {
 				ss.First = ss.Last
+				ss.lastNeedsUpdate = false
```

Comment:
This one threw me a bit before realising that `firstNeedsUpdate` is already cleared above, maybe a comment would help to make it clearer.

Reply:
Done

Reply:
Thanks!